x-pack/plugins/security/server/routes/user_profile/update.ts

```diff
@@ -47,7 +48,26 @@ export function defineUpdateUserProfileDataRoute({
       }
 
       const currentUser = getAuthenticationService().getCurrentUser(request);
+
       const userProfileData = request.body;
+      const imageDataUrl = userProfileData.avatar.imageUrl;
+      const matches = imageDataUrl.match(/^data:([A-Za-z-+\/]+);base64,(.+)$/);
+      if (!matches || matches.length !== 3) {
+        return response.customError({
+          body: 'Unsupported media type',
+          statusCode: 415,
+        });
+      }
+
+      const [, mimeType] = matches;
+
+      if (!IMAGE_FILE_TYPES.includes(mimeType)) {
```

Comment:
Should we also constrain the image size (automatically resize) as we do via the UI?
https://github.com/elastic/kibana/blob/079db3e7ebcbed934c39d0efe67821317fe9eaec/x-pack/plugins/security/public/account_management/user_profile/utils.ts#L63C8-L63C8

I did notice that at some point the payload max size kicks in, which would limit the incoming image size, but not to same degree that the UI resizes.

> {
>   "statusCode": 413,
>   "error": "Request Entity Too Large",
>   "message": "Payload content length greater than maximum allowed: 1048576"
> }

Reply:
You can see that if I upload via API, the image is much more clear because it is not resized:
<img width="266" alt="Screenshot 2023-12-28 at 1 24 26 PM" src="https://github.com/elastic/kibana/assets/103939324/42b43463-4b15-4d2a-859e-e6b36fa15622">

But if I upload via the UI, it gets resized:
<img width="264" alt="Screenshot 2023-12-28 at 1 24 02 PM" src="https://github.com/elastic/kibana/assets/103939324/27ec1056-1c4b-4153-8345-911462500047">

Reply:
Good idea! As image resizing is done via the Image and Canvas APIs on the browser, we'll need to use libraries to do the same for the node server which means adding a new [dependency](https://github.com/Automattic/node-canvas). Where does this dependency go? Top level package.json?

Reply:
Adding a separate issue for this. https://github.com/elastic/security/issues/1868